### DISCUSSION TITLE
Is there already a way to persist the redux state?

### DISCUSSION THREAD

**Discussion by Legion2** [reduxkotlin/redux-kotlin#128] [Ideas]
I am looking for something similar to [redux-persist](https://github.com/rt2zz/redux-persist).

**Comment by mpetuska** [comment]
Not at the moment, however you could simply subscribe to a store and persist state manually on each update to achieve something similar.

**Reply by Legion2** [reply]
But then the state must also be restored on application start. Restoring may take some time and the application should wait and not read the state in an inconsistent state.

**Reply by mpetuska** [reply]
You could hack around that by starting an app with initial state (as if the state was never saved) and then asynchronously reading a state and dispatching an action that would fully replace (or merge) the state with whatever you had persisted.

**Reply by mpetuska** [reply]
BTW I appreciate it's not the answer you were looking for and agree that it'd be a nice addition to redux-kotlin toolkit. Let me know if you'd like to contribute a module for this.
